demo/installer/x86_64/install.sh

```diff
@@ -126,7 +126,11 @@ create_demo_msdos_partition()
 }
 
 eval $create_demo_partition $blk_dev
-demo_dev="${blk_dev}$demo_part"
+if [[${onie_dev} == sd*]]; then
+    demo_dev="${blk_dev}$demo_part"
+else
+    demo_dev="${blk_dev}p$demo_part"
+fi
```

Comment:
I think this test will be better if you check explicitly for "mmcblk" in the device name.  It is the only blk device type that requires special handling.

As it is written right now this will have a problem on the virtual machine implementation, whose blk device looks like /dev/vda.

Also all the ONIE scripts are written for busybox's ASH shell, not BASH.  So please do not use BASH-isms like "[[" and "==".

ASH is very nearly identical to Debian's (and Ubuntu's) default system interpreter "dash" shell.  See the man page dash(1) -- http://manpages.ubuntu.com/manpages/lucid/man1/sh.1.html